kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -147,10 +150,99 @@ public static Map<String, Variant<?>> buildIpv4Settings(NetworkProperties props,
+                        .toNMSettingIP6ConfigAddrGenMode(ipv6AddressGenerationMode).toInt32()));
+            }
+
+            Optional<String> privacy = props.getOpt(String.class, "net.interface.%s.config.ip6.privacy", deviceId);
+            if (privacy.isPresent()) {
+                KuraIp6Privacy ip6Privacy = KuraIp6Privacy.fromString(privacy.get());
+                settings.put("ip6-privacy",
+                        new Variant<>(KuraIp6Privacy.toNMSettingIP6ConfigPrivacy(ip6Privacy).toInt32()));
+            }
+
+        } else if (ip6ConfigMethod.equals(KuraIp6ConfigurationMethod.DHCP)) {
+
+            settings.put(NM_SETTINGS_IPV6_METHOD, new Variant<>("dhcp"));
+
+        } else if (ip6ConfigMethod.equals(KuraIp6ConfigurationMethod.MANUAL)) {
+
+            settings.put(NM_SETTINGS_IPV6_METHOD, new Variant<>("manual"));
+
+            String address = props.get(String.class, "net.interface.%s.config.ip6.address", deviceId);
+            Short prefix = props.get(Short.class, "net.interface.%s.config.ip6.prefix", deviceId);
+
+            Map<String, Variant<?>> addressEntry = new HashMap<>();
+            addressEntry.put("address", new Variant<>(address));
+            addressEntry.put("prefix", new Variant<>(new UInt32(prefix)));
+
+            if (ip6Status.equals(KuraIpStatus.ENABLEDWAN)) {
+                Optional<String> gateway = props.getOpt(String.class, "net.interface.%s.config.ip6.gateway", deviceId);
+                gateway.ifPresent(gatewayAddress -> settings.put("gateway", new Variant<>(gatewayAddress)));
+            }
+
+            List<Map<String, Variant<?>>> addressData = Arrays.asList(addressEntry);
+            settings.put("address-data", new Variant<>(addressData, "aa{sv}"));
+
+        } else {
+            throw new IllegalArgumentException(
+                    String.format("Unsupported IPv6 address generation mode: \"%s\"", ip6ConfigMethod));
+        }
+
+        if (ip6Status.equals(KuraIpStatus.ENABLEDLAN)) {
+            settings.put("ignore-auto-dns", new Variant<>(true));
+            settings.put("ignore-auto-routes", new Variant<>(true));
+
+        } else if (ip6Status.equals(KuraIpStatus.ENABLEDWAN)) {
+            Optional<List<String>> dnsServers = props.getOptStringList("net.interface.%s.config.ip6.dnsServers",
+                    deviceId);
+            if (dnsServers.isPresent()) {
+                settings.put("dns", new Variant<>(convertIp6(dnsServers.get()), "aay"));
+                settings.put("ignore-auto-dns", new Variant<>(true));
+            }
+
+            Optional<Integer> wanPriority = props.getOpt(Integer.class, "net.interface.%s.config.ip6.wan.priority",
+                    deviceId);
+            if (wanPriority.isPresent()) {
+                Long supportedByNM = wanPriority.get().longValue();
+                settings.put("route-metric", new Variant<>(supportedByNM));
+            }
```

Comment:
Use `Optional.ifPresent` instead of branching everytime. This way Sonar shouldn't complain about complexity

```suggestion
            wanPriority.ifPresent(value -> 
                settings.put("route-metric", new Variant<>(value.longValue()))
            )
```